deploy/examples/create-external-cluster-resources.py

```diff
@@ -632,7 +632,8 @@ def __init__(self, arg_list=None):
         self.MIN_USER_CAP_PERMISSIONS = {
             "mgr": "allow command config",
             "mon": "allow r, allow command quorum_status, allow command version",
-            "osd": "allow rwx pool={0}.rgw.meta, "
+            "osd": "profile rbd-read-only, "
```

Comment:
For upgrade cases, you need to update the

client.healthchecker in the list https://github.com/rook/rook/issues/12931#issuecomment-1729865450
And also need to test the upgrade scenario.

Reply:
@parth-gr I added the changes, but could you help me with the upgrade steps? also, upstream and downstream upgrades are different steps, right?

Reply:
Create the caps with older versions of script, 
For upgrading use newer version  of script,
run the script with `python3 sript.py--upgrade` flag,

And see the permissions are getting upgraded from lower version of script to newer version

Reply:
Downstream is pretty much same.

But try to check with upgrade you are able to use your functionality

Reply:
Also check the upgrade works, I remember somewhat we should only add new permission to the end of the caps, but we can try this too

Reply:
@parth-gr does that matter? I do see unit test is failing but not sure why

Reply:
yes, it is just unit test we need to fix